### ISSUE TITLE
Plots.savefig is 3-6x slower in Julia 1.9 Master vs 1.8 rc3

### ISSUE THREAD

**Issue by truedichotomy** [JuliaLang/julia#46082]
I'm running a plotting script using Plots.jl with plotly backend. When comparing the performance of the code in Julia 1.9 master vs 1.8 rc3, the code is 3-6x slower in 1.9. The difference in performance mainly comes from Plots.savefig(). It seems that the issue lies in GC.

For example, here is a clip of the output timing in 1.8:

```
"1 Virginia"
  1.066138 seconds (3.89 M allocations: 1.025 GiB, 24.39% gc time, 1.51% compilation time)
"2 North Carolina"
  0.604263 seconds (3.13 M allocations: 850.525 MiB, 10.29% gc time)
"3 West Virginia"
  0.362660 seconds (2.12 M allocations: 578.525 MiB, 5.82% gc time)
"4 Delaware"
  0.154254 seconds (995.94 k allocations: 282.826 MiB)
```

```
julia> versioninfo()
Julia Version 1.8.0-rc3
Commit 33f19bcbd25 (2022-07-13 19:10 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.3.0)
  CPU: 8 × Apple M1
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, apple-m1)
  Threads: 1 on 4 virtual cores
```

and here is the same code run in 1.9:

```
"1 Virginia"
  3.588847 seconds (3.87 M allocations: 1.025 GiB, 76.38% gc time, 0.34% compilation time)
"2 North Carolina"
  2.897771 seconds (3.13 M allocations: 851.070 MiB, 77.34% gc time)
"3 West Virginia"
  2.037664 seconds (2.11 M allocations: 578.860 MiB, 79.17% gc time)
"4 Delaware"
  0.951250 seconds (993.95 k allocations: 282.965 MiB, 78.04% gc time)
```

```
julia> versioninfo()
Julia Version 1.9.0-DEV.994
Commit 29586ef30f* (2022-07-17 20:12 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.5.0)
  CPU: 8 × Apple M1
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.5 (ORCJIT, apple-m1)
  Threads: 1 on 4 virtual cores
```

**Comment by KristofferC** [comment]
Please provide enough information so that the slowdown can be reproduced.

**Comment by ViralBShah** [comment]
Can you provide a test file and a script that will help reproduce this?

**Comment by truedichotomy** [comment]
Sorry it's taken me a while to get back to this. However I just tested the same code on the latest Master and the performance has been restored.

```
"1 Virginia"
  0.857857 seconds (3.95 M allocations: 1.033 GiB, 11.91% gc time, 3.32% compilation time: 62% of which was recompilation)
"2 North Carolina"
  0.682614 seconds (3.14 M allocations: 853.685 MiB, 10.67% gc time)
"3 West Virginia"
  0.443301 seconds (2.12 M allocations: 580.650 MiB, 11.92% gc time)
"4 Delaware"
  0.205517 seconds (997.09 k allocations: 283.786 MiB, 9.31% gc time)
```

```
julia> versioninfo()
Julia Version 1.9.0-DEV.1019
Commit 028e9ff8d3* (2022-07-20 15:15 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.5.0)
  CPU: 10 × Apple M1 Pro
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.5 (ORCJIT, apple-m1)
  Threads: 1 on 8 virtual cores
```

**Comment by truedichotomy** [comment]
Very odd, the issue does not appear on my M1 Pro mac but does show up on my M1 mac. Not sure what's going on but looking for an opportunity to dig deeper into this.

**Comment by giordano** [comment]
But you still haven't shared the code, to let _others_ try and reproduce the issue.

**Comment by IanButterworth** [comment]
Is it repeatable on each machine? 

It could be that the use of E and P cores is somehow different.

If they work for you, the efforts in https://github.com/m-j-w/CpuId.jl/issues/46 might shed light on what the core ids are.

**Comment by truedichotomy** [comment]
This issue MIGHT have been unique to my machine. I rebooted my MacBook Air and it acted like it was still finishing installing something from the update to macOS 12.5. At the same time, I did a fresh new install of Julia 1.9 master. After all that, my script ran at normal speed again and the GC activity resumed to normal. The reason I haven't shared the code is that it's part of a much larger code base and the plotting is the final step of it which was the step the slowed down. I haven't had the time to reproduce the problem with minimal reproducible example. Until this problem pops again consistently, I think we can consider this issue closed. Thanks.
